include/quda_internal.h

```diff
@@ -18,6 +18,21 @@
 #include <qmp.h>
 #endif
 
+// these are helper macros used to enable spin-1, spin-2 and spin-4 building blocks as needed
+#if defined(GPU_WILSON_DIRAC) || defined(GPU_DOMAIN_WALL_DIRAC) || defined(GPU_CLOVER_DIRAC)                           \
+  || defined(GPU_TWISTED_MASS_DIRAC) || defined(GPU_TWISTED_CLOVER_DIRAC) || defined(GPU_NDEG_TWISTED_MASS_DIRAC)      \
+  || defined(GPU_CLOVER_HASENBUSCH_TWIST) || defined(GPU_COVDEV)
+#define NSPIN4
+#endif
+
+#if defined(GPU_MULTIGRID)
+#define NSPIN2
+#endif
+
+#if defined(GPU_STAGGERED_DIRAC)
```

Comment:
Should there be a define related to the laplace operator too? In the chance someone is linking against QUDA to use it, you'd need `NSPIN1` blas routines...

Reply:
We have both `nSpin=1` and `nSpin=4` Laplace operators, so these are conditionally compiled depending on whether staggered and Wilson are enabled, respectively.